diff --git a/src/java/database/DatabaseManager.java b/src/java/database/DatabaseManager.java
index daae634..43bd57a 100644
--- a/src/java/database/DatabaseManager.java
+++ b/src/java/database/DatabaseManager.java
@@ -56,12 +56,11 @@ public class DatabaseManager
             // A weak compare and swap is used to help with potential memory contention in the case it is not initialized.
             if (status == DATABASE_UNINITIALIZED && INIT_STATE.weakCompareAndSet(DATABASE_UNINITIALIZED, DATABASE_INITIALIZING)) {
                 DatabaseManager.createErrorLogsTable();
+                DatabaseManager.createDataDescriptionTable();
                 DatabaseManager.createDataValueTable();
                 DatabaseManager.createManualDataValueTable();
-                DatabaseManager.createDataValueTable();
                 DatabaseManager.createManualDataNamesTable();
                 DatabaseManager.createUserTable();
-                
                 // Need to fill parameter table as this is potentially first time running
                 // At least on the actual server...
                 DataReceiver.getParameters().map(DataParameter::getName).subscribeOn(Schedulers.io()).blockingSubscribe(DatabaseManager::insertDataName);
diff --git a/src/java/servlets/AdminServlet.java b/src/java/servlets/AdminServlet.java
index f567439..6b8e112 100644
--- a/src/java/servlets/AdminServlet.java
+++ b/src/java/servlets/AdminServlet.java
@@ -55,6 +55,7 @@ static {
         session.setAttribute("user", new common.User());
         common.User admin = (common.User) session.getAttribute("user");
         String action = request.getParameter("action");
+        log("Action is: " + action);
         
         DatabaseManager.init();
         /*
